src/cascadia/TerminalApp/SettingsTab.cpp

```diff
@@ -0,0 +1,177 @@
+        _CreateContextMenu();
+        _CreateIcon();
+    }
+
+    // Method Description:
+    // - Initializes a TabViewItem for this Tab instance.
+    // Arguments:
+    // - <none>
+    // Return Value:
+    // - <none>
+    void SettingsTab::_MakeTabViewItem()
+    {
+        TabViewItem(::winrt::MUX::Controls::TabViewItem{});
+        TabViewItem().Header(winrt::box_value(Title()));
+    }
+
+    // Method Description:
+    // - Focus the settings UI
+    // Arguments:
+    // - focusState: The FocusState mode by which focus is to be obtained.
+    // Return Value:
+    // - <none>
+    void SettingsTab::Focus(WUX::FocusState focusState)
+    {
+        _focusState = focusState;
+
+        if (_focusState != FocusState::Unfocused)
+        {
+            Content().Focus(focusState);
+        }
+    }
+
+    WUX::FocusState SettingsTab::FocusState() const noexcept
+    {
+        return _focusState;
+    }
+
+    // Method Description:
+    // - Set the icon on the TabViewItem for this tab.
+    // Arguments:
+    // - <none>
+    // Return Value:
+    // - <none>
+    winrt::fire_and_forget SettingsTab::_CreateIcon()
+    {
+        auto weakThis{ get_weak() };
+
+        co_await winrt::resume_foreground(TabViewItem().Dispatcher());
+
+        if (auto tab{ weakThis.get() })
+        {
+            auto fontFamily = winrt::WUX::Media::FontFamily(L"Segoe MDL2 Assets");
+            auto glyph = L"\xE713"; // This is the Setting icon (looks like a gear)
+
+            // The TabViewItem Icon needs MUX while the IconSourceElement in the CommandPalette needs WUX...
+            IconSource(GetFontIcon<winrt::WUX::Controls::IconSource>(fontFamily, 12, glyph));
+            TabViewItem().IconSource(GetFontIcon<winrt::MUX::Controls::IconSource>(fontFamily, 12, glyph));
+
+            // Update SwitchToTab command's icon
+            SwitchToTabCommand().IconSource(GetFontIcon<winrt::WUX::Controls::IconSource>(fontFamily, 12, glyph));
+        }
+    }
+
+    // Method Description:
+    // - Gets the title string of the settings UI
+    // Arguments:
+    // - <none>
+    // Return Value:
+    // - the title string of the last focused terminal control in our tree.
+    winrt::hstring SettingsTab::GetActiveTitle() const
+    {
+        // TODO: This _could_ change depending on what page of the settings UI is open/focused?
+        return Title();
+    }
+
+    // Method Description:
+    // - Prepares this tab for being removed from the UI hierarchy
+    void SettingsTab::Shutdown()
+    {
+        // TODO: Does/Will the settings UI need some shutdown procedures?
+        Content(nullptr);
+        _ClosedHandlers(nullptr, nullptr);
+    }
+
+    // Method Description:
+    // - Creates a context menu attached to the tab.
+    // Currently contains elements allowing the user to close the selected tab
+    // Arguments:
+    // - <none>
+    // Return Value:
+    // - <none>
+    void SettingsTab::_CreateContextMenu()

```

Comment:
Hmm... Maybe this context menu should be abstracted differently. I'm looking at #7728 and we would have to duplicate that code here and in `TerminalTab`. Here's a few ideas:

1. Could we make `ITab` an abstract class instead of an interface? That way the context menu gets inherited by both (and all future tabs)
2. Should we move the responsibility of populating the context menu over to `TerminalPage`? That way it checks the type of tab we have, and populates it accordingly?

Reply:
I def agree with wanting to clean up the context menu creation in light of incoming additions in that PR and in the future. Here's another idea - I kinda want to make our own custom control named something like "TabContextFlyout". 
- Flyout menu items become xaml code - we won't need to have cpp code that creates our menu items. 
- Each tab type owns one of these controls.
- Each tab can hook itself up to the `Click` events of each of the control's menu items that they wish to use. Then we can turn each menu item's visibility off if the click handler isn't registered.

Reply:
You got me at "become xaml code" haha.

Reply:
Alright this might take a little longer than I thought. I'm gonna file a follow up issue for this so that I can get this PR in, work on the issue on the side and unblock myself for some data binding work.

Reply:
issue filed here: #7841